### PR TITLE
Promotion from AMD internal branch for 2022.Q2.2

### PR DESCRIPTION
**Pull request by WenqingLiAMD**
@kuhar

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5f12f8e2e62330bed0bec266d1ae34f43a400696
<details><summary>CTS tests (Failed: 0/186422)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36645/65226 (56.2%)</li>
<li>Failed: 0/65226 (0.0%)</li>
<li>Not Supported: 28581/65226 (43.8%)</li><li>Warnings: 0/65226 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30002/55970 (53.6%)</li>
<li>Failed: 0/55970 (0.0%)</li>
<li>Not Supported: 25968/55970 (46.4%)</li><li>Warnings: 0/55970 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37804/65226 (58.0%)</li>
<li>Failed: 0/65226 (0.0%)</li>
<li>Not Supported: 27422/65226 (42.0%)</li><li>Warnings: 0/65226 (0.0%)</li>
</ul></details>


</ul></details>

**Review by kuhar — APPROVED** [review]
LGTM modulo the formatting change pointed out by @Flakebi

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -6853,7 +6859,7 @@ bool SPIRVToLLVM::transMetadata() {
           computeMode.workgroupSizeY = workgroupSizeY;
           computeMode.workgroupSizeZ = workgroupSizeZ;
           getBuilder()->setComputeShaderMode(computeMode);
-          // clang-format on
+        // clang-format on
```
[the remark this answers is not in the document]
Comment (WenqingLiAMD) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp:6862]: > clang-format complains about the removed indentation here

Hi Flakebi, the indentation is from the result of my local clang-format-11 tool, I'm not clear why the clang-format  complains, do you have any idea?

[the remark this answers is not in the document]
Comment (Flakebi) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp:6862]: Weird, our GitHub CI uses clang-format-11 too. I tested locally with clang-format from LLVM 10, 11 and 13, and all of them produce the indented result (same as the GitHub CI).

For reference, GitHub has this command-line: `git diff ${{ github.base_ref }} -U0 --no-color -- '**/*.cpp' '**/*.cc' '**/*.h' '**/*.hh' | clang-format-diff-11 -p1` (where `${{ github.base_ref }}` is something like `HEAD~1` to format the last 1 commit)

**Comment by amdrexu** [comment]
@WenqingLiAMD Please rebase the PR and check if you need to address the issue pointed out by Sebastian. Thanks.

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (WenqingLiAMD) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp:6862]: Thanks Flakebi, I can reproduce the issue locally: if we do clang-format once, we get current indentation result, but if we continue to check the clang-format, the issue will be observed. I'll fix it in next promotion.

**Pull request closed without merging**